metrics_sdk/lib/opentelemetry/sdk/metrics/configuration_patch.rb

```diff
@@ -10,12 +10,57 @@ module Metrics
       # The ConfiguratorPatch implements a hook to configure the metrics
       # portion of the SDK.
       module ConfiguratorPatch
+        def add_metric_reader(metric_reader)
+          @metric_readers << metric_reader
+        end
+
         private
 
-        # The metrics_configuration_hook method is where we define the setup process
-        # for metrics SDK.
+        def initialize
+          super
+          @metric_readers = []
+        end
+
+        # The metrics_configuration_hook method is where we define the setup process for the metrics SDK.
         def metrics_configuration_hook
           OpenTelemetry.meter_provider = Metrics::MeterProvider.new(resource: @resource)
+          configure_metric_readers
+        end
+
+        def configure_metric_readers
+          readers = @metric_readers.empty? ? wrapped_metric_exporters_from_env.compact : @metric_readers
+          readers.each { |r| OpenTelemetry.meter_provider.add_metric_reader(r) }
+        end
+
+        def wrapped_metric_exporters_from_env
+          exporters = ENV.fetch('OTEL_METRICS_EXPORTER', 'otlp')
+          exporters.split(',').map do |exporter|
+            case exporter.strip
+            when 'none' then nil
+            when 'console'
+              OpenTelemetry.meter_provider.add_metric_reader(
+                Metrics::Export::PeriodicMetricReader.new(
+                  exporter: Metrics::Export::ConsoleMetricPullExporter.new
+                )
+              )
+            when 'otlp'
```

Comment:
@mwear @xuan-cao-swi - [The specs state that OTLP should be the default exporter or `OTEL_METRICS_EXPORTER`](https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#exporter-selection), so I updated the tests to reflect this and installed the OTLP exporter in the test group of the Gemile.

However, the OTLP exporter brings in the protobuf dependencies, making the entire metrics_sdk test suite incompatible with JRuby. 

It seems like the tracing test suite gets around this by avoiding any mention of the OTLP exporter in the tests, even though the spec also expects OTLP to be the default exporter for that signal (the tests use zipkin instead).

I don't want to skip JRuby for all tests in the metrics_sdk. 

My current thoughts for solutions are: 
1. conditionally install the OTLP exporter in the test environment if JRuby is present and skip the related tests if JRuby is present 
2. omit the OTLP exporter from the tests entirely, and bring in an alternate exporter for the tests, following the pattern for traces (in-memory metrics exporter seems like the best candidate, even though it's not part of the spec)
3. something else...? 

What do y'all think would be the best next step?

Reply:
We talked about this during the SIG today. 

For testing, use the in-memory exporter when you can.
For this particular case, where it's testing this default, just skip that on JRuby.